app/models/concerns/licenses/gateway.rb

```diff
@@ -16,4 +22,33 @@ def trial_ending?
   def subscription_ending?
     active? && subscribed_until && (subscribed_until - License::NOTICE_PERIOD) <= Time.zone.now
   end
+
+  def price_per_month
+    LICENSE_PLANS[auditors_limit].to_f
+  end
+
+  def generate_payment_request
+    return subscription_id if subscription_id.present?
+
+    result = MercadoPagoClient.create_preapproval_payment self
+
+    update(
+      subscription_id:  result[:id],
+      subscribed_until: result[:valid_until]
+    )
+  end
+
+  def check_license_status
+    return if cancelled?
+
+    result = MercadoPagoClient.last_payment_for self.id
+
+    if result[:status] == :paid && result[:approved_at] >= 35.days.ago
+      update status: :active, paid_until: result[:approved_at] + 1.month
+    elsif result[:status] == :in_process
+      update status: :active, paid_until: 2.days.from_now
+    else
+      update status: :unpaid
```

Comment:
Podría ser, para que se entienda mejor (ya entendí lo que hacías) algo así:
```suggestion
      update status: :unpaid unless unpaid?
```